backend/tag/signals.py

```diff
@@ -0,0 +1,54 @@
+from django.core.exceptions import ValidationError
+from django.db.models.signals import m2m_changed
+from django.dispatch import receiver
+
+from .models import DOCS_PER_TAG_LIMIT, Tag
+
+
+def tag_exceeds_maximum(tag: Tag, n_added: int):
+    return (tag.tagged_docs.count() + n_added) > DOCS_PER_TAG_LIMIT
```

Comment:
I think I mentioned this before, but this limit is not that useful if we also allow users to
- make a infinite number of `Tag`s
- search for the union of any number of tags.

It could make more sense to impose the limit on a (User, Corpus) scope, rather than (User, Tag), as I think that is more what we care about.

Reply:
making this into an issue